src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -170,30 +170,32 @@ private static byte[] TryGetByteArrayFromStream(Stream data)
         }
 
         /// <summary>
-        /// If the MemoryStream was created with publiclyVisible=true, then we can access its buffer
+        /// If the MemoryStream was created with publiclyVisible=true, then we can try to access its buffer
         /// directly and save allocations in StreamReader. The input MemoryStream is not closed on exit.
         /// </summary>
         /// <returns>True if a byte array could be created.</returns>
         private static bool TryGetByteArrayFromMemoryStream(MemoryStream data, out byte[] buffer)
         {
             Debug.Assert(data.Position == 0);
 
-            try
+            if (PortableShim.MemoryStream.TryGetBuffer != null)
             {
-                if (PortableShim.MemoryStream.GetBuffer != null)
```

Comment:
@svick The `GetBuffer` method is designed to throw when the offset is not 0.  Every ctor, save one, which is provided an index sets internal state to throw on a `GetBuffer` call.  There is one overload where you can bypass this behavior but it's not commonly used.

I don't want my comments to be interpreted as me being happy with this method.  Quite the opposite.  I'm still somewhat burned that I had to write it this way.  But given that we have to target .Net Core there is no good option here:

1. Neither of the methods are available on both .Net Core + .Net 4.5.2.  
2. The `TryGetBuffer` method is preferable but not available everywhere we have to run. 
3. The `GetBuffer` method works except in extreme cases and is available via reflection on most platforms.